src/libraries/System.Private.CoreLib/src/System/IO/Strategies/FileStreamAwaitableProvider.cs

```diff
@@ -0,0 +1,251 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.Runtime.ExceptionServices;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks.Sources;
+
+namespace System.IO.Strategies
+{
+    internal sealed partial class AsyncWindowsFileStreamStrategy : WindowsFileStreamStrategy
+    {
+        /// <summary>
+        /// Type that ensures we return a ValueTask when asynchronous work is scheduled when reading/writing inside AsyncWindowsFileStreamStrategy.
```

Comment:
nit: I think the type is self-describing (it's a nested type inside `AsyncWindowsFileStreamStrategy` and it implements `IValueTaskSource<int>` and `IValueTaskSource`). for me the most important aspect is the allocations reduction and the possibility to do both read and write

```suggestion
        /// Type that helps us to reduce allocations for FileStream.ReadAsync and FileStream.WriteAsync.
```